sdk/confidentialledger/azure-confidentialledger/samples/get_and_verify_receipt.py

```diff
@@ -0,0 +1,136 @@
+    # automatically when the program exits.
+    with tempfile.TemporaryDirectory() as tempdir:
+        ledger_cert_file = os.path.join(tempdir, f"{ledger_id}.pem")
+        with open(ledger_cert_file, "w") as outfile:
+            outfile.write(ledger_certificate["ledgerTlsCertificate"])
+
+        print(
+            f"Ledger certificate has been written to {ledger_cert_file}. "
+            "It will be deleted when the script completes."
+        )
+
+        # Build a client through AAD
+        ledger_client = ConfidentialLedgerClient(
+            ledger_endpoint,
+            credential=DefaultAzureCredential(),
+            ledger_certificate_path=ledger_cert_file,
+        )
+
+        # Write a ledger entry and wait for the transaction to be committed.
+        try:
+            entry_contents = "Hello world!"
+            post_poller = ledger_client.begin_create_ledger_entry(
+                {"contents": entry_contents}
+            )
+            post_entry_result = post_poller.result()
+            transaction_id = post_entry_result["transactionId"]
+            print(
+                f"Wrote '{entry_contents}' to the ledger at transaction {transaction_id}."
+            )
+        except HttpResponseError as e:
+            if e.response != None:
+                print("Request failed: {}".format(e.response.json()))
+            else:
+                print("No response found")
+            raise
+
+        # Get a receipt for a ledger entry.
+        # A receipt can be retrieved for any transaction id to provide cryptographic proof of the
+        # contents of the transaction.
+        try:
+            print(
+                f"Retrieving a receipt for {transaction_id}. The receipt may be used to "
+                "cryptographically verify the contents of the transaction."
+            )
+            print(
+                "For more information about receipts, please see "
+                "https://microsoft.github.io/CCF/main/audit/receipts.html#receipts"
+            )
+            get_receipt_poller = ledger_client.begin_get_receipt(transaction_id)
+            get_receipt_result = get_receipt_poller.result()
+            print(f"Receipt for transaction id {transaction_id}: {get_receipt_result}")
+        except HttpResponseError as e:
+            if e.response != None:
+                print("Request failed: {}".format(e.response.json()))
+            else:
+                print("No response found")
+            raise
+
+        # Read content of service certificate file saved in previous step.
+        with open(ledger_cert_file, "r") as service_cert_file:
+            service_cert_content = service_cert_file.read()
+
+        try:
+            # Verify the contents of the receipt.
+            verify_receipt(get_receipt_result["receipt"], service_cert_content)
+            print(f"Receipt for transaction id {transaction_id} successfully verified")
+        except Exception:
```

Comment:
This will always be `ValueError` if the verification failed right? If so we can handle it specifically.

Reply:
Addressed in https://github.com/Azure/azure-sdk-for-python/pull/27265/commits/e558b7840ceebc54673364d5886280c06a7e1ad1